test/e2e-tests.sh

```diff
@@ -85,6 +85,18 @@ toggle_feature autocreate-cluster-domain-claims true config-network || fail_test
 go_test_e2e -timeout=2m ./test/e2e/domainmapping ${E2E_TEST_FLAGS} || failed=1
 toggle_feature autocreate-cluster-domain-claims false config-network || fail_test
 
+toggle_feature dataplane-trust enabled config-network || fail_test
+toggle_feature "logging.enable-request-log" true config-observability || fail_test
+toggle_feature "logging.request-log-template" 'TLS: {{.Request.TLS}}' config-observability || fail_test
+# with the current implementation, Activator is always in the request path, and needs to be restarted after configuring dataplane-trust
```

Comment:
Restart is necessary not only due to  "activator always in the path" but also config watch needs to toggle this tls server from non-tls server.
https://github.com/knative/serving/blob/60cb95fb4cbf5495b362d38b11e0a6a46015e833/cmd/activator/main.go#L284